client/src/nonblocking/quic_client.rs

```diff
@@ -282,13 +287,30 @@ impl QuicClient {
     }
 
     async fn _send_buffer_using_conn(
+        _tpu_address: SocketAddr,
         data: &[u8],
         connection: &NewConnection,
+        server_reply_channel: Option<BidirectionalChannelHandler>,
     ) -> Result<(), QuicError> {
-        let mut send_stream = connection.connection.open_uni().await?;
+        let create_bichannel = match &server_reply_channel {
+            Some(x) => !x.is_serving(),
+            None => false,
+        };
+        if create_bichannel {
+            let (mut send_stream, recv_stream) = connection.connection.open_bi().await?;
 
-        send_stream.write_all(data).await?;
-        send_stream.finish().await?;
+            send_stream.write_all(data).await?;
+            send_stream.finish().await?;
+            let server_reply_channel = server_reply_channel.unwrap();
+            server_reply_channel.start_serving(recv_stream);
+        } else {
+            let mut send_stream = connection.connection.open_uni().await?;
+
+            send_stream.write_all(data).await?;
+            send_stream.write_all(data).await?;
+            send_stream.finish().await?;
+            send_stream.finish().await?;
```

Comment:
why do you call both methods twice? before this wasn't needed

Reply:
fixed

Reply:
super wierd it is not updated in the UI but i can see the changes in the commit.